x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformIndexer.java

```diff
@@ -146,7 +146,7 @@ public TransformIndexer(
         this.checkpointProvider = ExceptionsHelper.requireNonNull(checkpointProvider, "checkpointProvider");
         this.auditor = transformServices.getAuditor();
         this.transformConfig = ExceptionsHelper.requireNonNull(transformConfig, "transformConfig");
-        this.progress = progress != null ? progress : new TransformProgress();
+        this.progress = transformProgress != null ? transformProgress : new TransformProgress();
```

Comment:
OK, this seemed like a bug, `progress` was assigning to itself, which was never set? We should set it to the passed value.

Reply:
That was broken in #75459, which is a 7.15 change, and explains why it hasn't caused a bug report from a user yet.

Reply:
😌